### PR TITLE
singleBlog

### PR DESCRIPTION
**Pull request by moatazalaa14**
i have error in fetching data but i dont know whats problem ? 😢😢  i will try to resolve it tomorrow

### CODE REVIEW

File: src/modules/blog/components/singleBlogPage/index.js [hunk 1]
```diff
@@ -1,7 +1,96 @@
...
+    blog: {},
+    status: false,
+  };
+  componentDidMount = () => {
+    getBlogs().then((res) => {
+      const blog = res.data.find(
+        (blog) => blog.id === this.props.match.params.id
+      );
+      if (blog) {
+        this.setState({ blog: { blog }, status: true });
```
Comment (AbudZohair) [inline src/modules/blog/components/singleBlogPage/index.js]: Refactor the setState of blog

File: src/modules/blog/components/singleBlogPage/index.js [hunk 2]
```diff
@@ -1,7 +1,96 @@
...
+                </ul>
+              </div>
+              <div className="mt-5 py-5 col-lg-9">
+                <p>{body}</p>
+              </div>
+            </section>
+          </div>
+        ) : (
+          <>
+            <Redirect to={"/not-found"} />
```
Comment (AbudZohair) [inline src/modules/blog/components/singleBlogPage/index.js]: Refactor your conditional rendering logic

File: src/modules/blog/components/singleBlogPage/index.js [hunk 3]
```diff
@@ -1,7 +1,96 @@
...
+                  <li>
+                    {this.state.blog.categories
+                      ? categories.map((categorie) => {
+                          return categorie.categoryName;
+                        })
+                      : null}
+                  </li>
+                  <li>{author}</li>
+                  <li>
+                    <a href={authorFacebook}>
```
Comment (AbudZohair) [inline src/modules/blog/components/singleBlogPage/index.js]: where is the to attribute

File: src/modules/blog/components/singleBlogPage/index.js [hunk 4]
```diff
@@ -1,7 +1,96 @@
...
+                        className={styles[`social-links`]}
+                      />
+                    </a>
+                    <a href="#">
+                      <FontAwesomeIcon
+                        icon={faLinkedin}
+                        className={styles[`social-links`]}
+                      />
+                    </a>
+                    <a href="#">
```
Comment (AbudZohair) [inline src/modules/blog/components/singleBlogPage/index.js]: the social links should be opened in a new tab

File: src/modules/blog/components/singleBlogPage/index.js [hunk 5]
```diff
@@ -1,7 +1,96 @@
...
+      );
+      if (blog) {
+        this.setState({ blog: { blog }, status: true });
+      }
+    });
+  };
   render() {
-    return <h1> Single Blog Page is ready ! </h1>;
+    const {
+      cover,
```
Comment (AbudZohair) [inline src/modules/blog/components/singleBlogPage/index.js:24]: where is the cover in the blogpost

File: src/modules/blog/components/singleBlogPage/index.js [hunk 6]
```diff
@@ -1,7 +1,96 @@
...
+            </header>
+            <section className="row ">
+              <div
+                className={`mt-5 py-5 col-lg-3 ${
+                  styles[`blog-personaldetails`]
+                }`}
+              >
+                <ul className={styles[`blog-personaldetails__list`]}>
+                  <li>
+                    {this.state.blog.categories
```
Comment (AbudZohair) [inline src/modules/blog/components/singleBlogPage/index.js:64]: Refactor this logic for Ui, semantics, and accessibility

**Review by AbudZohair — CHANGES_REQUESTED** [review]

**Comment by AbudZohair** [comment]
> i have error in fetching data but i dont know whats problem ? 😢😢 i will try to resolve it tomorrow

You don't have a fetching data error.

File: src/modules/blog/components/singleBlogPage/style.module.css [hunk 7]
```diff
@@ -0,0 +1,15 @@
+.blog-title{
+    font-size: 5rem;
+}
+.blog-created{
```
Comment (AbudZohair) [inline src/modules/blog/components/singleBlogPage/style.module.css:5]: use bootstrap classes

File: src/modules/blog/components/singleBlogPage/style.module.css [hunk 8]
```diff
@@ -0,0 +1,15 @@
+.blog-title{
```
Comment (AbudZohair) [inline src/modules/blog/components/singleBlogPage/style.module.css:1]: the heading (h1) should match the project style guide rules

**Comment by AbudZohair** [comment]
Refactor the UI of small-screen devices

File: src/modules/blog/components/singleBlogPage/index.js (hunk 2 above)
Comment (moatazalaa14) [reply src/modules/blog/components/singleBlogPage/index.js]: what is problem why i should change redirect ?does it have any side-effects ?

**Comment by moatazalaa14** [comment]
> > i have error in fetching data but i dont know whats problem ? 😢😢 i will try to resolve it tomorrow
> 
> You don't have a fetching data error.

Capture6565

**Comment by AbudZohair** [comment]
Ohh sorry for that add".. /.. /" at the beginning of the path of get request

File: src/modules/blog/components/singleBlogPage/index.js (hunk 2 above)
Comment (AbudZohair) [reply src/modules/blog/components/singleBlogPage/index.js]: With your logic
The redirection will happen once the component is mounted

another way is to Push to the history inside the componentDidMount

File: src/modules/blog/components/singleBlogPage/index.js [hunk 9]
```diff
@@ -1,7 +1,112 @@
...
+      title,
+      categories,
+      author,
+      authorFacebook,
+      body,
+      createdAt,
+    } = this.state.blog;
+    return (
+      <>
+        {this.state.status ? (
```
Comment (dvmhmdsd) [inline src/modules/blog/components/singleBlogPage/index.js]: What's `status` ?

File: src/modules/blog/components/singleBlogPage/index.js [hunk 10]
```diff
@@ -1,7 +1,112 @@
...
+                  </li>
+                  <br />
+                  <li className={styles[`blog-personaldetails__item`]}>
+                    <a href={authorFacebook} target="_blank">
+                      <FontAwesomeIcon
+                        icon={faFacebook}
+                        className={styles[`social-links`]}
+                      />
+                    </a>
+                    <a href="#" target="_blank">
```
Comment (dvmhmdsd) [inline src/modules/blog/components/singleBlogPage/index.js]: No fake links.. What's the purpose of these links here ? the user will have `facebook` link only and it's optional, so use `conditional rendering`

**Review by dvmhmdsd — CHANGES_REQUESTED** [review]
- When I navigate to the single blog page I see a blank page !

**Pull request merged**